### PR TITLE
add proposal data into CRC proposal and proposal tracking payload

### PR DESCRIPTION
**Pull request by RainFallsSilent**

### CODE REVIEW

File: core/types/payload/crcproposal.go [hunk 1]
```diff
@@ -89,28 +100,39 @@ func (p *CRCProposal) Data(version byte) []byte {
...
 	for _, v := range p.Budgets {
 		if err := v.Serialize(w); err != nil {
-			return err
+			return errors.New("failed to serialize Budgets")
 		}
 	}
+
 	if err := p.Recipient.Serialize(w); err != nil {
-		return errors.New("Recipient serialize failed")
+		return errors.New("failed to serialize CRSponsorDID")
```
Comment (Houshoupei84) [inline core/types/payload/crcproposal.go]: Recipient

File: core/types/payload/crcproposaltracking.go [hunk 2]
```diff
@@ -152,34 +160,40 @@ func (p *CRCProposalTracking) Serialize(w io.Writer, version byte) error {
...
 
 	if err = p.DocumentHash.Deserialize(r); err != nil {
-		return errors.New("the DocumentHash deserialize failed")
+		return errors.New("failed to deserialize DocumentHash")
+	}
+
+	p.DocumentData, err = common.ReadVarBytes(r, MaxProposalDataSize,
+		"documentData")
+	if err != nil {
+		return fmt.Errorf("failed to deserialize DraftData %s", err)
```
Comment (Houshoupei84) [inline core/types/payload/crcproposaltracking.go]: DocumentData

File: blockchain/txvalidator.go [hunk 3]
```diff
@@ -1603,21 +1603,29 @@ func (b *BlockChain) checkCRCProposalTrackingTransaction(txn *Transaction,
...
 		return errors.New("proposal status is not VoterAgreed")
 	}
 
+	// Check proposal tracking count.
 	if proposalState.TrackingCount >= b.chainParams.MaxProposalTrackingCount {
 		return errors.New("reached max tracking count")
 	}
 
+	// Check draft data of proposal tracking document.
+	dataHash := common.Uint256(common.Sha256D(cptPayload.DocumentData))
```
Comment (Houshoupei84) [inline blockchain/txvalidator.go]: maybe add  a DocumentHash() function to payload is more proper

File: core/types/payload/crcproposal.go (hunk 1 above)
Comment (RainFallsSilent) [reply core/types/payload/crcproposal.go]: fixed

File: core/types/payload/crcproposaltracking.go (hunk 2 above)
Comment (RainFallsSilent) [reply core/types/payload/crcproposaltracking.go]: fixed

File: blockchain/txvalidator.go (hunk 3 above)
Comment (RainFallsSilent) [reply blockchain/txvalidator.go]: added

**Pull request merged**